### PR TITLE
MM-48744 - enhance telemetry info when trial started from the learn more about trial modal

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
enhance telemetry info when trial started from the learn more about trial modal.

The telemetry info, once a trial has started from the "learn more about trial modal" will be:

Cloud:
- start_cloud_trial__learn_more_modal__onboarding
- start_cloud_trial__learn_more_modal__pricing_modal
- start_cloud_trial__learn_more_modal__main_menu_cloud_trial

Self hosted:
- start_trial__learn_more_modal__ onboarding
- start_trial__learn_more_modal__ pricing_modal
- start_trial__learn_more_modal__main_menu_cloud_trial

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48744

#### Related Pull Requests
n/a

#### Screenshots

Captura de Pantalla 2022-11-30 a las 19 21 20

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by johndavidlugtu** [comment]
@pvev would telemetry for activating a trial for each Pro/Enterprise feature in the system console be in another PR?

**Comment by pvev** [comment]
> @pvev would telemetry for activating a trial for each Pro/Enterprise feature in the system console be in another PR?

yes, the code that covers that scenario has been already merge https://github.com/mattermost/mattermost-webapp/pull/11634

**Comment by johndavidlugtu** [comment]
Oh awesome, thanks @pvev

**Comment by pvev** [comment]
> Oh awesome, thanks @pvev

yw! 👍

File: components/cloud_start_trial/cloud_start_trial_btn.tsx [hunk 1]
```diff
@@ -147,16 +147,25 @@ const CloudStartTrialButton = ({
...
             telemetryId,
         );
+
+        // on click will execute whatever action is sent from the invoking place, if nothing is sent, open the trial benefits modal
+        if (onClick) {
+            onClick();
+            return;
+        }
+
+        await openTrialBenefitsModal(updatedStatus);
```
[the remark this answers is not in the document]
Comment (pvev) [reply components/cloud_start_trial/cloud_start_trial_btn.tsx:168]: it wont, the return won't let the trial benefits modal to be opened. From the adldap upsell banner you define what you would expect to happen once the button is clicked, then that callback is executed and the return statement finish the execution.

**Review by julmondragon — APPROVED** [review]
LGTM! Thanks @pvev

File: components/cloud_start_trial/cloud_start_trial_btn.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (AGMETEOR) [reply components/cloud_start_trial/cloud_start_trial_btn.tsx:168]: Oh yeah, you're right.

**Pull request merged**